Source/Core/Common/FixedSizeQueue.h

```diff
@@ -45,6 +42,46 @@ class FixedSizeQueue
     tail = (tail + 1) % N;
   }
 
+  // Copies over an array, loops over if num is greater than max_size
+  void push_array(const T* t, size_t num)
+  {
+    size_t back_writable_num = std::min(tail_to_end(), num);
+    memcpy(&back(), t, sizeof(T) * back_writable_num);
```

Comment:
This is undefined behavior is T is not trivially copyable. (Also, this should be std::memcpy, not just memcpy)

Reply:
we can just put a static assert for that then?

Reply:
No, this class is used with non-trivially copyable types (like a pair of std::string and [whatever], and std::string is already not trivially copyable).

Reply:
Never mind, what I said made no sense.
Are you saying that it's undefined behaviour because the destructor on overwritten items is not being called?
Or because I'm not calling the copy constructor? Or both?

Reply:
Both. The memcpy itself is not UB but you'll almost certainly run into UB later, when you try to use those incorrectly copied strings, or when they get destructed and you get a double free, etc.

Reply:
There are 3 solutions I can think of:
-Make a different version which just falls back on "push()".
-Make a child class which only takes trivial types, where these methods are implemented.
-Put an assert if the type is non trivial.

Reply:
I opted for std::enable_if. Works great.